shared/middleware/net_sim_api.rb

```diff
@@ -281,3 +304,16 @@ def parse_table_map_from_query_string(query_string)
     end
   end
 end
+
+# Convert a query string of the form "id[]=1&id[]=2" into an array of integer
+# ids as expected by RedisTable.delete_many().  Noninteger ids in the query
+# are simply omitted from the result.
+def parse_ids_from_query_string(query_string)
+  CGI::parse(query_string)['id[]'].map do |x|
+    begin
+      Integer(x, 10)
+    rescue ArgumentError
+      nil
+    end
+  end.select {|x| not x.nil?}
```

Comment:
You could replace the select clause with just compact. 

Consider however the following, which avoids an array copy and is a bit simpler.  

[].tap do |ids| 
  CGI::parse(query_string)['id[]'].each do |id|
      ids << Integer(x, 10) rescue ArgumentError
  end
end